scality_manila_utils/smb_helper.py

```diff
@@ -0,0 +1,351 @@
+    """
+    Retrieve client details of an export.
+
+    :param export_name: name of export
+    :type export_name: string (unicode)
+    :param exports: all the defined shares in the Samba registry
+    :type exports: dictionary
+    :returns: string with export client details in json format
+    """
+
+    export = exports[export_name]
+    clients = dict((host, ["rw"]) for host in export['hosts allow'].split())
+
+    return json.dumps(clients)
+
+
+@ensure_environment
+def add_export(root_export, export_name, *args, **kwargs):
+    """
+    Add an export.
+
+    :param root_export: SOFS directory which holds the export points exposed
+        through manila
+    :type root_export: string (unicode)
+    :param export_name: name of export to add
+    :type export_name: string (unicode)
+    """
+
+    if not export_name or '/' in export_name:
+        raise ExportException('Invalid export name')
+
+    export_point = os.path.join(root_export, export_name)
+
+    create_cmd = [
+        'net', 'conf', 'addshare', export_name, export_point,
+        'writeable=y', 'guest_ok=y',
+    ]
+    parameters = {
+        'browseable': 'yes',
+        'create mask': '0755',
+        'hosts deny': '0.0.0.0/0',  # deny all by default
+        'hosts allow': '127.0.0.1',
+        'read only': 'no',
+    }
+
+    set_of_commands = [['net', 'conf', 'setparm', export_name,
+                        param, value] for param, value in parameters.items()]
+
+    with utils.elevated_privileges():
+        try:
+            os.mkdir(export_point)
+        except OSError as exc:
+            if exc.errno != errno.EEXIST:
+                raise
+            else:
+                log.debug("The share/directory %s already exists on SOFS",
+                          export_name)
+                exports = _get_defined_exports()
+                if export_name in exports:
+                    msg = ("Share '{0:s}' already defined in Samba "
+                           "registry.".format(export_name))
+                    raise ExportAlreadyExists(msg)
+
+        os.chmod(export_point, 0o0777)
+
+        subprocess.check_call(create_cmd)
+        for cmd in set_of_commands:
+            subprocess.check_call(cmd)
+
+
+@ensure_environment
+@ensure_export_exists
+def wipe_export(root_export, export_name, exports):
+    """
+    Remove an export.
+
+    The export point is not actually removed, but renamed with the prefix
+    "TRASH-".
+
+    :param root_export: SOFS directory which holds the export points exposed
+        through manila
+    :type root_export: string (unicode)
+    :param export_name: name of export to remove
+    :type export_name: string (unicode)
+    :param exports: all the defined shares in the Samba registry
+    :type exports: dictionary
+    """
+
+    export = exports[export_name]
+    export_path = os.path.join(root_export, export_name)
+    if export['hosts allow'] and export['hosts allow'] != '127.0.0.1':
```

Comment:
I want to raise a `ExportHasGrantsException` if `export['hosts allow']` is not a empty string and `export['hosts allow']` is not `127.0.0.1`.